test/unit/lookup_key_test.rb

```diff
@@ -24,7 +24,7 @@ def test_path2match_single_domain_path
     key   = ""
     value = ""
     as_admin do
-      key   = LookupKey.create!(:key => "ntp", :path => "domain", :puppetclass => Puppetclass.first)
+      key   = LookupKey.create!(:key => "ntp", :path => "domain", :puppetclass => Puppetclass.first, :key_type => "string")
```

Comment:
not sure I follow - how will respond_to be helpful?
also - don't understand what the bug is - if key_type or validator_type are nil, the lookup key isn't going to run the caster nor the validator. this behaviour didn't change.